vault/logical_passthrough.go

```diff
@@ -218,8 +219,19 @@ func (b *PassthroughBackend) handleList(
 		return nil, err
 	}
 
+	// A list of an actual key returns "" in the list, which can cause nasty
+	// things downstream with JSON conversion, including in Go. So, prepend the
+	// path and let users do what they wish.
+	retKeys := make([]string, len(keys))
+	for i, k := range keys {
+		retKeys[i] = req.MountPoint + req.Path + k
```

Comment:
Do we want to always return the full path or just the path relative to the request?